Add support for create_many to SQLite

AdeelK93 opened the pull request:
## Change Summary

* #994
* Now that every provider supports `create_many`, we do not need to distinguish between database providers
* However, not every provider supports `skip_duplicates`, so this is now tracked in tests

## Checklist

- [ ] Unit tests for the changes exist
- [ ] Tests pass without significant drop in coverage
- [ ] Documentation reflects changes where applicable
- [ ] Test snapshots have been [updated](https://prisma-client-py.readthedocs.io/en/latest/contributing/contributing/#snapshot-tests) if applicable

## Agreement

By submitting this pull request, I confirm that you can use, modify, copy and redistribute this contribution, under the terms of your choice.

RobertCraigie commented on databases/utils.py:
```diff
@@ -20,7 +20,7 @@
     'json_arrays',
     'raw_queries',
-    'create_many',
+    'skip_duplicates',
```
nit: I think this should be `create_many_skip_duplicates` and would be great to rename the test files as well :)

RobertCraigie commented on databases/sync_tests/test_batch.py:
```diff
@@ -250,16 +250,16 @@ def test_execute_raw(client: Prisma, raw_queries: RawQueries) -> None:
-        pytest.skip('The create_many method is supported by the current behaviour')
+    """Cannot call create_many with skip_duplicates on databases that do not support it"""
+    if 'skip_duplicates' not in config.unsupported_features:
+        pytest.skip('The create_many skip_duplicates argument is supported by the current behaviour')
```
nit: this message was slightly wrong before 😅
```suggestion
        pytest.skip('The create_many skip_duplicates argument is supported by the current database')
```

RobertCraigie commented on databases/tests/test_batch.py:
```diff
@@ -266,16 +266,16 @@ async def test_execute_raw(client: Prisma, raw_queries: RawQueries) -> None:
-        pytest.skip('The create_many method is supported by the current behaviour')
+    """Cannot call create_many with skip_duplicates on databases that do not support it"""
+    if 'skip_duplicates' not in config.unsupported_features:
+        pytest.skip('The create_many skip_duplicates argument is supported by the current behaviour')
```
same here 😅 
```suggestion
        pytest.skip('The create_many skip_duplicates argument is supported by the current database')
```

RobertCraigie commented on src/prisma/generator/templates/actions.py.jinja:
```diff
@@ -232,8 +232,8 @@ class {{ model.name }}Actions(Generic[{{ ModelType }}]):
         """
-        if self._client._active_provider == 'sqlite':
-            raise errors.UnsupportedDatabaseError('sqlite', 'create_many()')
+        if skip_duplicates and self._client._active_provider in ['mongodb', 'sqlserver', 'sqlite']:
```
nit: can you share the Prisma docs that say this is unsupported? might be a good idea to add that link in a comment here as well? (applies to `client.py.jinja` as well)

RobertCraigie replied:
separately, this is a tiny nit but it'd be more efficient for this to be a set
```suggestion
        if skip_duplicates and self._client._active_provider in {'mongodb', 'sqlserver', 'sqlite'}:
```
it might also make sense to extract this to `_constants.py` as well? I don't feel strongly

RobertCraigie reviewed (commented):
Awesome, thank you! Just minor nits :)

AdeelK93 commented:
thanks @RobertCraigie, i've addressed your feedback. could you please re-run the tests? it’s the same transient error as last time breaking all the windows tests (incomplete read while downloading node binary)

AdeelK93 commented:
I'm confused, why are windows tests failing on this PR, [but passing on my fork](https://github.com/AdeelK93/prisma-client-py/actions/runs/10321758123)? 😕

RobertCraigie commented:
I think it's just unlucky network issues :(

```
ile "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\http\client.py", line 626, in _safe_read

    chunk = self.fp.read(min(amt, MAXAMOUNT))

  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\socket.py", line 704, in readinto

    return self._sock.recv_into(b)

  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\ssl.py", line 1242, in recv_into

    return self.read(nbytes, buffer)

  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\ssl.py", line 1100, in read

    return self._sslobj.read(len, buffer)

ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

nodeenv installation failed; You may want to try installing `nodejs-bin` as it is more reliable.
```

RobertCraigie commented on docs/reference/operations.md:
````diff
@@ -36,6 +33,10 @@ users = await db.user.create_many(
 ```
+
+!!! warning
+    the `skip_duplicates` argument is not supported for SQLite
````
nit: this should also mention MongoDB & SQLServer

RobertCraigie reviewed (approved):
Great work, thank you!
